Automattic/wp-calypso
# Update a product's variations when attributes are changed.

This PR adds a reducer that fires on `WOOCOMMERCE_EDIT_PRODUCT_ATTRIBUTE` and updates the variation edits state based on what new variation types/attributes are available. 

If an existing variation is no longer present in our list of allowed variations, it is dropped. If a variation does not exist yet, but is in our list of allowed variations, a new variation is created. Otherwise, variations (and their associated data) are preserved.

To Test:
* Run `make test` and make sure all tests pass.
* You can also open `http://calypso.localhost:3000/store/:site/products/add` and start typing variation types in. Once you type in a type and value, you should see the variation edits Redux state update with valid variations if you have Redux dev tools.
